ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionsSpec.scala

```diff
@@ -426,6 +431,73 @@ private[dao] trait JdbcLedgerDaoTransactionsSpec extends OptionValues with Insid
     }
   }
 
+  // Test case for #6698
+  it should "return empty source when offset range is from the future" in {
+    val commands: Vector[(Offset, LedgerEntry.Transaction)] = Vector.fill(3)(singleCreate)
+    val beginOffsetFromTheFuture = nextOffset();
+    val endOffsetFromTheFuture = nextOffset()
+
+    for {
+      _ <- commands.traverse(x => store(x))
+
+      result <- ledgerDao.transactionsReader
+        .getFlatTransactions(
+          beginOffsetFromTheFuture,
+          endOffsetFromTheFuture,
+          Map(alice -> Set.empty[Identifier]),
+          verbose = true)
+        .runWith(Sink.seq)
+
+    } yield {
+      extractAllTransactions(result) shouldBe empty
+    }
+  }
+
+  it should "return all transactions in the specified offset range when iterating with pageSize = 2" in {
```

Comment:
```suggestion
  it should "return all transactions in the specified offset range when iterating with gaps in the offsets assigned to events and a page size that ensures a page ends in such a gap" in {
```
Is it about the page size, the gaps, or a mix of the two? If the gaps are relevant, it's probably worth mentioning them in the test case description.